pydantic_core/core_schema.py

```diff
@@ -1447,16 +1492,74 @@ def __call__(self, __input_value: Any, __info: ValidationInfo) -> Any:  # pragma
         ...
 
 
+class ModelFieldValidatorFunction(Protocol):
+    def __call__(self, __input_value: Any, __info: ModelFieldValidationInfo) -> Any:  # pragma: no cover
+        ...
+
+
+class ModelFieldValidatorFunctionSchema(TypedDict):
+    call: Required[ModelFieldValidatorFunction]
+    type: Literal['method']
+
+
+class ValidatorFunctionSchema(TypedDict):
+    call: Required[ValidatorFunction]
+    type: Literal['function']
+
+
 class FunctionSchema(TypedDict, total=False):
+    function: Required[Union[ModelFieldValidatorFunctionSchema, ValidatorFunctionSchema]]
```

Comment:
@samuelcolvin let me know how you feel about this. I'm solving the problem of "states that can't exist" by using another level of indirection 🙄

Reply:
please can we keep `type` as the first attribute of the type-dict, just for consistency.

In terms of the type, I'm marginally in favour of

```py
ModelFieldValidatorFunctionSchema = tuple[Literal['method'], ModelFieldValidatorFunction]
ValidatorFunctionSchema = tuple[Literal['function'], ValidatorFunction]
```